### ISSUE TITLE
Error with pytest 8.3.4 and Python 3.11

### ISSUE THREAD

**Issue by alexmojaki** [15r10nk/inline-snapshot#139]
1. Clone https://github.com/pydantic/logfire
2. `uv sync --python 3.11 --upgrade-package pytest`
3. `uv run --python 3.11 pytest -k test_inspect_arguments`

```pytb
=============================================================================== inline snapshot ===============================================================================
Traceback (most recent call last):
  File "/Users/alex/work/logfire/.venv/bin/pytest", line 8, in <module>
    sys.exit(console_main())
             ^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 201, in console_main
    code = main()
           ^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/main.py", line 330, in pytest_cmdline_main
    return wrap_session(config, _main)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/main.py", line 318, in wrap_session
    config.hook.pytest_sessionfinish(
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/logging.py", line 868, in pytest_sessionfinish
    return (yield)
            ^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 124, in _multicall
    teardown.send(result)  # type: ignore[union-attr]
    ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/terminal.py", line 903, in pytest_sessionfinish
    self.config.hook.pytest_terminal_summary(
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/terminal.py", line 925, in pytest_terminal_summary
    return (yield)
            ^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/_pytest/warnings.py", line 132, in pytest_terminal_summary
    return (yield)
            ^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/pytest_plugin.py", line 251, in pytest_terminal_summary
    for change in snapshot._changes():
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_inline_snapshot.py", line 868, in _changes
    yield from self._value._get_changes()
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_inline_snapshot.py", line 511, in _get_changes
    yield from check(self._old_value, self._ast_node, self._new_value)
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_inline_snapshot.py", line 380, in check
    yield from check(
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_inline_snapshot.py", line 434, in check
    yield from check(old_value[key], node, new_value[key])
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_inline_snapshot.py", line 434, in check
    yield from check(old_value[key], node, new_value[key])
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_inline_snapshot.py", line 494, in check
    and self._token_of_node(old_node) != new_token
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/work/logfire/.venv/lib/python3.11/site-packages/inline_snapshot/_utils.py", line 115, in __eq__
    return ast.literal_eval(self.string) == ast.literal_eval(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/.pyenv/versions/3.11.10/lib/python3.11/ast.py", line 110, in literal_eval
    return _convert(node_or_string)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/.pyenv/versions/3.11.10/lib/python3.11/ast.py", line 109, in _convert
    return _convert_signed_num(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alex/.pyenv/versions/3.11.10/lib/python3.11/ast.py", line 83, in _convert_signed_num
    return _convert_num(node)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/alex/.pyenv/versions/3.11.10/lib/python3.11/ast.py", line 74, in _convert_num
    _raise_malformed_node(node)
  File "/Users/alex/.pyenv/versions/3.11.10/lib/python3.11/ast.py", line 71, in _raise_malformed_node
    raise ValueError(msg + f': {node!r}')
ValueError: malformed node or string on line 1: <ast.JoinedStr object at 0x119c42a10>
```

It's trying to call `ast.literal_eval("f'log {GLOBAL_VAR} {local_var}'")`.

**Comment by 15r10nk** [comment]
I was able to reproduce this. The minimal example which I found is:

``` python
from inline_snapshot import snapshot 
snapshot(f'')
```

**Comment by 15r10nk** [comment]
fixed in 0.14.1

**Comment by alexmojaki** [comment]
Thanks!
